### PR TITLE
Remove needless dep

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -74,15 +74,10 @@ jobs:
         key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
 
     - name: Compile
-      uses: actions-rs/cargo@v1
```
Comment (bjorn3) [inline .github/workflows/ci.yaml]: This action creates annotations for errors.

Comment (matklad) [reply .github/workflows/ci.yaml]: yeah, but I personally found that to be distracting rather than valuable

File: .github/workflows/rustdoc.yaml [hunk 2]
```diff
@@ -27,10 +27,8 @@ jobs:
...
 
     - name: Deploy Docs
-      uses: peaceiris/actions-gh-pages@v2.6.0-rc0
-      env:
-        ACTIONS_DEPLOY_KEY: ${{ secrets.ACTIONS_DEPLOY_KEY }}
-        PUBLISH_BRANCH: gh-pages
-        PUBLISH_DIR: ./target/doc
-      with:
-        forceOrphan: true
+      uses: peaceiris/actions-gh-pages@364c31d33bb99327c77b3a5438a83a357a6729ad # v3.4.0
```
Comment (Veetaha) [inline .github/workflows/rustdoc.yaml:30]: @matklad could you please elaborate why this commit sha works, but newer versions don't?

Comment (matklad) [reply .github/workflows/rustdoc.yaml:30]: Newer versions might work as well, but I don't want to randomly pull unknown new code into CI job which runs with high privileges.

Comment (matklad) [reply .github/workflows/rustdoc.yaml:30]: Basically, it's a poor-man Cargo.lock

**Pull request merged**